### PR TITLE
Channel Emote Manipulation

### PR DESCRIPTION
**Pull request by AnatoleAM**
Making the "Add To / Remove From Channel" button functional

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -40,8 +40,11 @@ export default defineComponent({
 
 		// Fetch authed user
 		const { onResult } = useQuery<GetUser>(GetUser, { id: "@me" });
-		onResult(({ data: { user } }) => {
-			store.commit("SET_USER", user);
+		onResult((res) => {
+			if (!res.data) {
```
Comment (TroyDota) [inline src/App.vue:44]: if they are not logged in?

File: src/assets/scss/components/button.scss [hunk 2]
```diff
@@ -8,45 +8,51 @@
 	color: white;
 	transition: background-color 200ms ease;
 
-	@include themify() {
-		&[color="primary"] {
-			$color: themed("primary");
+	&:not([disabled]) {
```
Comment (TroyDota) [inline src/assets/scss/components/button.scss:11]: what is the idea behind this?

File: src/assets/scss/emote-page.scss [hunk 3]
```diff
@@ -1,11 +1,11 @@
 @import "./themes.scss";
 
 .emote-page {
-    padding-top: 2.5em;
     margin-right: 5%;
     margin-left: 5%;
 
     .heading-bar {
+		margin-top: 1.5em;
```
Comment (TroyDota) [inline src/assets/scss/emote-page.scss:8]: bad indent

File: src/views/EmotePage/EmotePage.vue [hunk 4]
```diff
@@ -178,14 +195,16 @@ export default defineComponent({
 		});
 
 		return {
-			emote,
+			emote: emote as unknown as Emote,
```
Comment (TroyDota) [inline src/views/EmotePage/EmotePage.vue:198]: dont do this, with changing the type def, rather use Ref<Emote>

**Review by TroyDota — CHANGES_REQUESTED** [review]

File: src/views/EmotePage/EmotePage.vue [hunk 5]
```diff
@@ -74,24 +70,33 @@
 				<div class="is-content-block">TODO</div>
 			</div>
 		</template>
-		<template v-else-if="!partial">Loading...</template>
+		<template v-else-if="loading">Loading...</template>
+		<template v-else>
+			<div class="emote-unknown">
+				<NotFoundPage></NotFoundPage>
```
Comment (TroyDota) [inline src/views/EmotePage/EmotePage.vue:76]: On components that have no children, it looks neater when doing the 
```
<NotFoundPage />
```
over the extended syntax

**Review by TroyDota — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 1 above)
Comment (AnatoleAM) [reply src/App.vue:44]: Just temporary until I fully figure out auth

File: src/assets/scss/components/button.scss (hunk 2 above)
Comment (AnatoleAM) [reply src/assets/scss/components/button.scss:11]: Just disables the button

File: src/views/EmotePage/EmotePage.vue [hunk 6]
```diff
@@ -178,14 +191,16 @@ export default defineComponent({
 		});
 
 		return {
-			emote,
+			emote: ref(emote),
```
Comment (TroyDota) [inline src/views/EmotePage/EmotePage.vue:194]: Why are u wrapping it in another ref?

Comment (AnatoleAM) [reply src/views/EmotePage/EmotePage.vue:194]: idk FeelsDankMan

**Pull request merged**